### PR TITLE
[example] Adding an example of training dreambooth with the new booster API

### PR DESCRIPTION
**Pull request by MaruyamaAya**
## 📌 Checklist before creating the PR

- [x] I have created an issue for this PR for traceability
- [x] The title follows the standard format: `[doc/gemini/tensor/...]: A concise description`
- [x] I have added relevant tags if possible for us to better distinguish different PRs

## 🚨 Issue number

> Link this PR to your issue with words like fixed to automatically close the linked issue upon merge
>
> e.g. `fixed #1234`, `closed #1234`, `resolved #1234`

## 📝 What does this PR do?
adding an example of training dreambooth with the new booster API
> Summarize your work here.
> if you have any plots/diagrams/screenshots/tables, please attach them here.

## 💥 Checklist before requesting a review

- [x] I have linked my PR to an issue ([instruction](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue))
- [x] My issue clearly describes the problem/feature/proposal, with diagrams/charts/table/code if possible
- [x] I have performed a self-review of my code
- [x] I have added thorough tests.
- [x] I have added docstrings for all the functions/methods I implemented

## ⭐️ Do you enjoy contributing to Colossal-AI?

- [x] 🌝 Yes, I do.
- [ ] 🌚 No, I don't.

Tell us more if you don't enjoy contributing to Colossal-AI.

### CODE REVIEW

File: examples/images/dreambooth/train_dreambooth_colossalai.py [hunk 1]
```diff
@@ -642,36 +667,24 @@ def collate_fn(examples):
...
-                torch_unet = get_static_torch_model(unet)
                 if local_rank == 0:
-                    pipeline = DiffusionPipeline.from_pretrained(
-                        args.pretrained_model_name_or_path,
-                        unet=torch_unet,
-                        revision=args.revision,
-                    )
                     save_path = os.path.join(args.output_dir, f"checkpoint-{global_step}")
-                    pipeline.save_pretrained(save_path)
+                    booster.save_model(unet, os.path.join(save_path, "diffusion_pytorch_model.bin"))
```
Comment (ver217) [inline examples/images/dreambooth/train_dreambooth_colossalai.py:672]: This function should be called on all processes.

File: examples/images/dreambooth/train_dreambooth_colossalai_lora.py [hunk 2]
```diff
@@ -664,27 +693,24 @@ def collate_fn(examples):
 
             if global_step % args.save_steps == 0:
                 torch.cuda.synchronize()
-                torch_unet = get_static_torch_model(unet)
                 if local_rank == 0:
                     save_path = os.path.join(args.output_dir, f"checkpoint-{global_step}")
-                    torch_unet = torch_unet.to(torch.float32)
-                    torch_unet.save_attn_procs(save_path)
+                    booster.save_model(unet, os.path.join(save_path, "diffusion_pytorch_model.bin"))
```
Comment (ver217) [inline examples/images/dreambooth/train_dreambooth_colossalai_lora.py:698]: Same problem as the above

File: examples/images/dreambooth/test_ci.sh [hunk 3]
```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
+set -xe
+pip install -r requirements.txt
+
+HF_DATASETS_OFFLINE=1
+TRANSFORMERS_OFFLINE=1
+DIFFUSERS_OFFLINE=1
+
+for plugin in "torch_ddp" "torch_ddp_fp16" "gemini" "low_level_zero"; do
+  torchrun --nproc_per_node 4 --master_port=25641 train_dreambooth_colossalai.py \
```
[the remark this answers is not in the document]
Comment (MaruyamaAya) [reply examples/images/dreambooth/test_ci.sh:10]: This shell command is inherited from the last version of colossalai.sh 
No change has been done.

File: examples/images/dreambooth/train_dreambooth_colossalai.py (hunk 1 above)
Comment (MaruyamaAya) [reply examples/images/dreambooth/train_dreambooth_colossalai.py]: fixed

File: examples/images/dreambooth/train_dreambooth_colossalai_lora.py (hunk 2 above)
Comment (MaruyamaAya) [reply examples/images/dreambooth/train_dreambooth_colossalai_lora.py]: fixed

File: examples/images/dreambooth/test_ci.sh (hunk 3 above)
[the remark this answers is not in the document]
Comment (ver217) [reply examples/images/dreambooth/test_ci.sh:10]: Please add `--standalone` and remove `--master_port=25641`. In CI environment, fixed port has risk of conflict.

**Review by ver217 — APPROVED** [review]

**Comment by FrankLeeeee** [comment]
@MaruyamaAya only 4 GPUs are available on the CI machine, please restrict `--nproc_per_node` to be maximum 4.

**Pull request merged**